### PR TITLE
Identity and Identity auth resource

### PR DESCRIPTION
**Pull request by akhilmhdh**
The PR adds a new resources for management of

1. Identity
2. Identity universal auth
3. Identity universal auth client secret
4. Aws auth
5. Azure auth
6. Gcp auth
7. K8s auth

Before merging this PR, [api changes pr](https://github.com/Infisical/infisical/pull/1999) needs to be merged.

### CODE REVIEW

**Comment by akhilmhdh** [comment]
@maidul98 you can start testing this. Don't merge yet. 

There is a small terraform edgecase in which when its `optional` and `computed`, the changes keeps triggering. I need a way to resolve that

**Review by maidul98 — COMMENTED** [review]
Made some small changes here and there and left some comments

File: internal/provider/resource/identity.go [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+				Description: "The name for the identity",
+				Required:    true,
+			},
+			"org_id": schema.StringAttribute{
+				Description:   "The ID of the organization for the identity",
+				Required:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.RequiresReplace()},
+			},
+			"role": schema.StringAttribute{
+				Description: "The role for the identity",
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply internal/provider/resource/identity.go:55]: Good one. Yah for now ill update the description also

File: internal/provider/resource/identity.go [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+				Required:    true,
+			},
+			"id": schema.StringAttribute{
+				Description:   "The ID of the identity",
+				Computed:      true,
+				PlanModifiers: []planmodifier.String{stringplanmodifier.UseStateForUnknown()},
+			},
+
+			"auth_mode": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply internal/provider/resource/identity.go:64]: Yes

File: internal/provider/resource/identity_aws_auth.go [hunk 3]
```diff
@@ -0,0 +1,350 @@
...
+		} else {
+			resp.Diagnostics.AddError(
+				"Error reading identity aws auth",
+				"Couldn't read identity aws auth from Infiscial, unexpected error: "+err.Error(),
+			)
+			return
+		}
+	}
+
+	updateAwsAuthStateByApi(ctx, resp.Diagnostics, &state, &identityAwsAuth)
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply internal/provider/resource/identity_aws_auth.go:260]: That update is not api update rather its a reusable function used to update the tf state based on read data from API.

Read from terraform is used to sync the external entity with tf state when the entity changes externally without tf knowing about it. Its a way for synchronization of state. Update in tf is when you modify the tf value

File: internal/provider/resource/identity_kubernetes_auth.go [hunk 4]
```diff
@@ -0,0 +1,373 @@
...
+				MarkdownDescription: "A long-lived service account JWT token for Infisical to access the [TokenReview API](https://kubernetes.io/docs/reference/kubernetes-api/authentication-resources/token-review-v1/) to validate other service account JWT tokens submitted by applications/pods. This is the JWT token obtained from step 1.5.",
+				Required:            true,
+			},
+			"kubernetes_ca_certificate": schema.StringAttribute{
+				Description:         "The PEM-encoded CA cert for the Kubernetes API server. This is used by the TLS client for secure communication with the Kubernetes API server.",
+				MarkdownDescription: "The PEM-encoded CA cert for the Kubernetes API server. This is used by the TLS client for secure communication with the Kubernetes API server.",
+				Optional:            true,
+				Computed:            true,
+			},
+			"allowed_service_account_names": schema.StringAttribute{
```
[the remark this answers is not in the document]
Comment (akhilmhdh) [reply internal/provider/resource/identity_kubernetes_auth.go:84]: Noted

**Review by maidul98 — APPROVED** [review]
lgtm

**Pull request merged**